mitodl/ocw-hugo-projects
# feat: create a new external-resource type (collection)

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/3359

### Description (What does it do?)

This PR creates a new collection called `External Resources`. The entities in this collection will represent our external links.

### Screenshots (if appropriate):

<img width="1792" alt="Screenshot 2024-02-02 at 4 55 13 PM" src="https://github.com/mitodl/ocw-hugo-projects/assets/71316217/02e7cfdc-2ca1-4bba-be18-5790589ae6e8">

<img width="1792" alt="Screenshot 2024-02-02 at 4 55 25 PM" src="https://github.com/mitodl/ocw-hugo-projects/assets/71316217/66ae6973-7918-42c8-bcdb-f0e1dabc2648">

<img width="1792" alt="Screenshot 2024-01-29 at 3 00 00 PM" src="https://github.com/mitodl/ocw-hugo-projects/assets/71316217/91fa6b6f-77f3-4d20-922d-ae8529129992">

### How can this be tested?

1. Navigate to your local `ocw-studio` projects.
2. Replace the contents of `ocw-studio/localdev/configs/default-course-config.yml` with the content of `ocw-hugo-projects/ocw-course-v2/ocw-studio.yaml`.
3. Run
    ```sh
    docker compose exec web ./manage.py override_site_config -c localdev/configs/default-course-config.yml -s ocw-course-v2
    ```
6. Open [Studio](http://localhost:8043).
7. Open a course.
8. Expect to see an `External Resources` link in the left nav.
9. Open the `External Resources` tab.
10. Add a new external resource.
11. Expect the previous step to succeed.
12. Open/Create a page.
13. Add a link to the external resource via the `Add Link` button in the CKEditor UI.
14. Expect the previous step to succeed.
15. Select `Menu` from the left navigation.
16. Add a reference to the external resource you created in the menu.
17. Expect the previous step to succeed.



### Checklist:
- [ ] New shortcode functionality is merged.
